### PR TITLE
clarify error when no project.json in the path

### LINKED ISSUE #356: dotnet pack?
I was planning to do separate issues but something is seriously wrong here and needs to be fixed:
Dotnet compile help for dotnet pack
```
> E:\oss\cli\artifacts\win7-x64\stage2\bin\dotnet-pack.exe --help
.NET Compiler

Usage: dotnet compile [arguments] [options]
```

Help or some useful message expected, this found:
```
>dotnet-pack
System.IO.FileNotFoundException: Could not find file 'E:\oss\cli\project.json'.
File name: 'E:\oss\cli\project.json'
   at System.IO.Win32FileStream.Init(String path, FileMode mode, FileAccess access, FileShare share, Int32 bufferSize, FileOptions options, SECURITY_ATTRIBUTES secAttrs)
   at System.IO.Win32FileSystem.Open(String fullPath, FileMode mode, FileAccess access, FileShare share, Int32 bufferSize, FileOptions options, FileStream parent)
   at System.IO.FileStream.Init(String path, FileMode mode, FileAccess access, FileShare share, Int32 bufferSize, FileOptions options)
   at System.IO.FileStream..ctor(String path, FileMode mode, FileAccess access, FileShare share)
   at Microsoft.DotNet.ProjectModel.ProjectReader.GetProject(String projectFile, ICollection`1 diagnostics)
   at Microsoft.DotNet.ProjectModel.ProjectReader.GetProject(String projectFile)
   at Microsoft.DotNet.ProjectModel.ProjectContext.<CreateContextForEachFramework>d__28.MoveNext()
   at System.Linq.Enumerable.First[TSource](IEnumerable`1 source)
   at Microsoft.DotNet.Tools.Compiler.Program.BuildPackage(String path, String configuration, String outputValue, String intermediateOutputValue) in E:\oss\cli\src\Microsoft.DotNet.Tools.Pack\Program.cs:line 75
   at Microsoft.DotNet.Tools.Compiler.Program.<>c__DisplayClass0_0.<Main>b__0() in E:\oss\cli\src\Microsoft.DotNet.Tools.Pack\Program.cs:line 54
   at Microsoft.Dnx.Runtime.Common.CommandLine.CommandLineApplication.Execute(String[] args) in C:\Users\kwicher.REDMOND\.dnx\packages\Microsoft.Extensions.CommandLineUtils.Sources\1.0.0-rc2-16248\shared\CommandLineApplication.cs:line 263
   at Microsoft.DotNet.Tools.Compiler.Program.Main(String[] args) in E:\oss\cli\src\Microsoft.DotNet.Tools.Pack\Program.cs:line 59
```

We should probably write down some spec and implement it correctly

### PR DESCRIPTION
**Pull request by anurse**
also correct help information

fixes #356 

/cc @krwq @Sridhar-MS @brthor @davidfowl

### CODE REVIEW

File: src/Microsoft.DotNet.Tools.Pack/Program.cs [hunk 1]
```diff
@@ -48,6 +48,17 @@ public static int Main(string[] args)
                     path = Directory.GetCurrentDirectory();
                 }
 
+                if(!path.EndsWith(Project.FileName))
```
Comment (krwq) [inline src/Microsoft.DotNet.Tools.Pack/Program.cs]: nit: space after if

**Comment by krwq** [comment]
:shipit:

File: src/Microsoft.DotNet.Tools.Pack/Program.cs (hunk 1 above)
Comment (davidfowl) [inline src/Microsoft.DotNet.Tools.Pack/Program.cs]: everytime @anurse :smile:

File: src/Microsoft.DotNet.Tools.Pack/Program.cs [hunk 2]
```diff
@@ -48,6 +48,17 @@ public static int Main(string[] args)
...
+                {
+                    path = Path.Combine(path, Project.FileName);
+                }
+
+                if(!File.Exists(path))
```
Comment (troydai) [inline src/Microsoft.DotNet.Tools.Pack/Program.cs]: Here too.

File: src/Microsoft.DotNet.Tools.Pack/Program.cs (hunk 1 above)
Comment (anurse) [inline src/Microsoft.DotNet.Tools.Pack/Program.cs]: if only there was some tool I could run post-commit to reformat, say a `dotnet fmt` tool ;P

**Pull request merged**